core/src/test/java/org/elasticsearch/common/io/stream/BytesStreamsTests.java

```diff
@@ -622,4 +624,53 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeBoolean(value);
         }
     }
+
+    public void testWriteMapWithConsistentOrder() throws IOException {
+        Map<String, Object> map = new HashMap<>();
+        map.put("gWrgS", randomAsciiOfLength(5));
```

Comment:
Why hardcode these test keys?

For the `HashMap` default **capacity** is `16`, and **threshold** is 12(`DEFAULT_LOAD_FACTOR(0.75) * DEFAULT_INITIAL_CAPACITY(16)`).

for these keys, their slots should be:

| Key   | Slot |
|-------|------|
| gWrgS | 10   |
| HLRYi | 4    |
| HyKnF | 12   |

and for `Map<String, Object> reOrderMap = new HashMap<>(map);` it will recalculate capacity size and threshold, capacity: 8, threshold: 6,

and the generated slots should be:

| Key   | Slot |
|-------|------|
| gWrgS | 2    |
| HLRYi | 4    |
| HyKnF | 4 (collision!!! )    | 

when met **collision**, it will append this key into the existed node. 

for these test keys, when met collision,this two maps order will not equal.

but these two map generated **bytes** still should be equal.